### PR TITLE
Updated ExternalSchemaCodegen.bat to take a project path argument

### PR DESCRIPTION
**Pull request by joshuahuburn**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Describe your changes here.

#### Release note
REQUIRED: Add a release note to the `##Unreleased` section of CHANGELOG.md. You can find guidance for writing useful release notes [here](../SpatialGDK/Extras/internal-documentation/how-to-write-good-release-notes.md). Documentation changes are exempt from this requirement.

#### Tests
How did you test these changes prior to submitting this pull request?
What automated tests are included in this PR?

STRONGLY SUGGESTED: How can this be verified by QA?

#### Documentation
How is this documented (for example: release note, upgrade guide, feature page, in-code documentation)?

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: SpatialGDK/Build/Scripts/ExternalSchemaCodegen.bat [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 exit /b 1
 
 :START
 
 call :MarkStartOfBlock "Setup variables"
   pushd %~dp0\..\..\..
     set GDK_FOLDER=%cd%
-    pushd ..\..\..
+    pushd "%3"
       set GAME_FOLDER=%cd%
```
Comment (m-samiec) [inline SpatialGDK/Build/Scripts/ExternalSchemaCodegen.bat]: I thought using %cd% dynamically required delayed_expansion? Or is that just for nested loops?

File: SpatialGDK/Build/Scripts/ExternalSchemaCodegen.bat [hunk 2]
```diff
@@ -1,17 +1,18 @@
 @echo off 
 
-IF NOT "%~2"=="" IF "%~3"=="" GOTO START
+IF NOT "%~3"=="" IF "%~4"=="" GOTO START
```
Comment (m-samiec) [inline SpatialGDK/Build/Scripts/ExternalSchemaCodegen.bat]: CAPS

File: SpatialGDK/Build/Scripts/ExternalSchemaCodegen.bat [hunk 3]
```diff
@@ -76,4 +80,4 @@ exit /b 0
 
 :MarkEndOfBlock
 echo Finished: %~1
-exit /b 0
+exit /b 0
```
Comment (jared-improbable) [inline SpatialGDK/Build/Scripts/ExternalSchemaCodegen.bat]: Nit: missing newline at end of file

File: SpatialGDK/Build/Scripts/ExternalSchemaCodegen.bat [hunk 4]
```diff
@@ -1,26 +1,29 @@
...
 
 call :MarkStartOfBlock "Setup variables"
+  pushd %cd%\%1
+    set GAME_FOLDER=%cd%
+  popd
   pushd %~dp0\..\..\..
     set GDK_FOLDER=%cd%
-    pushd ..\..\..
-      set GAME_FOLDER=%cd%
+    pushd %GAME_FOLDER%
```
Comment (alastairdglennie) [inline SpatialGDK/Build/Scripts/ExternalSchemaCodegen.bat]: This pushd and its popd can actually be removed I think

File: SpatialGDK/Build/Scripts/ExternalSchemaCodegen.bat (hunk 1 above)
Comment (joshuahuburn) [reply SpatialGDK/Build/Scripts/ExternalSchemaCodegen.bat]: Only when inside an if statement or loop where the value of the variable can change and be using within the block.

**Pull request merged**